### PR TITLE
feat: update docker compose

### PR DESCRIPTION
**Pull request by epsjunior**
# Add Localnet Version Validation and WebDriver Service Migration

## Summary
Implements version validation to prevent users from initializing GenLayer CLI with localnet versions older than the minimum compatible version, and migrates Docker infrastructure from WebRequest to WebDriver service. This enhancement includes semantic version comparison, clear error messaging, updated Docker configuration, and comprehensive test coverage to improve developer experience and prevent compatibility issues.

### 🚀 New Features
- **Version Validation**: Validates localnet version against minimum compatible version during initialization
- **Semantic Version Comparison**: New `compareVersions` method with proper semver logic
- **Enhanced CLI Description**: Updated `--localnet-version` option to display minimum version requirement
- **Clear Error Messages**: Specific feedback when incompatible versions are detected
- **WebDriver Service**: Migrated from WebRequest to dedicated WebDriver service for improved web automation

### 🔧 Infrastructure Updates
- **Updated Compatible Version**: Bumped `localnetCompatibleVersion` from `v0.55.0` to `v0.65.0`
- **WebDriver Migration**: Replaced `webrequest` service with `webdriver` service in Docker configuration
- **Environment Configuration**: Updated environment variables from WebRequest to WebDriver setup
  - Changed `WEBREQUESTPORT`, `WEBREQUESTSELENIUMPORT`, `WEBREQUESTPROTOCOL`, `WEBREQUESTHOST` 
  - To `WEBDRIVERHOST` and `WEBDRIVERPORT` for cleaner configuration

### 🏗️ Implementation Details
- **Version Comparison Logic**: Handles versions with/without "v" prefix and different part counts
- **Integration with Existing Flow**: Works seamlessly with `normalizeLocalnetVersion` method
- **Smart Validation**: Skips version check when "latest" is specified
- **Service Architecture**: Added `compareVersions` to `SimulatorService` following established patterns
- **Interface Extension**: Updated `ISimulatorService` with proper TypeScript typing

### 📁 File Structure
```
src/
├── commands/general/
│   ├── init.ts             # Version validation logic in InitAction
│   └── index.ts           # Updated CLI option description
├── lib/services/
│   └── simulator.ts       # compareVersions method implementation
└── lib/interfaces/
    └── ISimulatorService.ts # Interface extension

Infrastructure:
├── .env.example            # Updated WebDriver configuration
├── docker-compose.yml      # Replaced webrequest with webdriver service
└── src/lib/config/simulator.ts # Updated compatible version
```

### 🐳 Docker Configuration Changes
- **WebDriver Service**: New `yeagerai/genlayer-genvm-webdriver:0.0.3` image
- **Simplified Setup**: Reduced environment variables and dependencies
- **Cleaner Architecture**: Removed complex WebRequest configuration
- **Required Containers**: Updated to use webdriver instead of webrequest

### 🧪 Testing
- **Simulator Service Tests** (`tests/services/simulator.test.ts`):
  - Version comparison with equal versions (with/without "v" prefix)
  - Less than and greater than comparisons
  - Mixed prefix scenarios and different part counts
  - Edge cases with various version formats

- **Init Action Tests** (`tests/actions/init.test.ts`):
  - Version validation failure for older versions
  - Successful flow for compatible and newer versions
  - "latest" version bypass behavior
  - Dynamic version references for future-proof testing

- **Command Tests** (`tests/commands/init.test.ts`):
  - Option description includes minimum version
  - Default value validation
  - CLI argument parsing verification

### 🔧 Usage
```bash
# Initialize with compatible version (succeeds)
genlayer init --localnet-version v0.65.0

# Initialize with newer version (succeeds)
genlayer init --localnet-version v0.66.0

# Initialize with older version (fails with clear error)
genlayer init --localnet-version v0.64.0
# Error: Localnet version v0.64.0 is not supported. Minimum required version is v0.65.0

# Initialize with latest (bypasses validation)
genlayer init --localnet-version latest

# View help to see minimum version requirement
genlayer init --help
# --localnet-version <localnetVersion>  Select a specific localnet version (minimum: v0.65.0)
```

### ✨ Code Quality
- **Future-Proof Testing**: Tests use dynamic version references instead of hardcoded values
- **Semantic Versioning**: Proper semver comparison logic with comprehensive edge case handling
- **TypeScript Integration**: Full type safety with interface extensions
- **Error Handling**: Graceful failure with `failSpinner` and early returns
- **Maintainable Architecture**: Version logic centralized in service layer
- **Clear Documentation**: Inline comments explaining version comparison logic

### 🎯 Version Validation Features
- **Flexible Input**: Handles versions with or without "v" prefix
- **Semantic Comparison**: Proper major.minor.patch comparison logic
- **Different Part Counts**: Handles "v1.0" vs "v1.0.0" comparisons
- **Clear Error Messages**: Specific feedback with actual vs required versions
- **Bypass Option**: "latest" always allowed for development flexibility

### 🔗 Dependencies
- Integrates with existing `SimulatorService` infrastructure
- Uses established `BaseAction` error handling patterns
- Compatible with current CLI command structure
- Maintains backward compatibility with all existing functionality
- **Updated Docker Stack**: Now uses WebDriver service for web automation

### 🛡️ Backward Compatibility
- **No Breaking Changes**: All existing commands work identically
- **Optional Validation**: Only applies when specific version is provided
- **Default Behavior**: Uses compatible version by default
- **Latest Support**: "latest" option continues to work as before
- **Docker Migration**: Seamless transition from WebRequest to WebDriver

---

**Testing**: ✅ All tests pass with comprehensive coverage  
**Type Safety**: ✅ Full TypeScript integration with proper interfaces  
**Documentation**: ✅ Clear inline documentation and updated CLI help  
**Architecture**: ✅ Follows established service and action patterns  
**User Experience**: ✅ Clear error messages and helpful CLI descriptions  
**Future-Proof**: ✅ Dynamic version references prevent test brittleness  
**Infrastructure**: ✅ Updated Docker stack with WebDriver service

## Summary by CodeRabbit

* **New Features**
  * Added support for a WebDriver service, replacing the previous WebRequest service in configuration and deployment.
  * Introduced version comparison functionality for localnet versions.

* **Improvements**
  * Enhanced initialization process to enforce a minimum supported localnet version ("v0.65.0").
  * Updated command-line option descriptions to display the minimum compatible localnet version.

* **Bug Fixes**
  * Ensured that initialization fails gracefully if an unsupported localnet version is specified.

* **Tests**
  * Added comprehensive tests for localnet version compatibility and version comparison logic.
  * Added tests to verify command-line option metadata reflects the correct minimum version.

### CODE REVIEW

File: src/lib/services/simulator.ts [hunk 1]
```diff
@@ -305,6 +305,24 @@ export class SimulatorService implements ISimulatorService {
...
+    for (let i = 0; i < Math.max(parts1.length, parts2.length); i++) {
+      const part1 = parts1[i] || 0;
+      const part2 = parts2[i] || 0;
+      
+      if (part1 < part2) return -1;
+      if (part1 > part2) return 1;
+    }
+    
+    return 0;
+  }
```
[the remark this answers is not in the document]
Comment (epsjunior) [reply src/lib/services/simulator.ts]:
```markdown

Applied the pre-release version fix - great catch on the `Number("0-alpha")` NaN issue. 

Didn't apply the other suggestions because:
- Input validation: Already handled by `normalizeLocalnetVersion()` which is called before this method
- "latest" handling: This method is never called when `localnetVersion === "latest"` - the validation is skipped entirely in that case
- Error throwing: Current architecture uses `failSpinner()` patterns rather than throwing errors
```

**Comment by cristiam86** [comment]
@epsjunior what happens if the user chooses a version that is higher than the latest version?

**Comment by epsjunior** [comment]
> @epsjunior what happens if the user chooses a version that is higher than the latest version?

It will attempt to download it, which is actually a good behavior — if the CLI isn’t updated for some reason, the user can still test a newer version.

I also added a new set of unit tests to cover these scenarios:
https://github.com/genlayerlabs/genlayer-cli/pull/237/files#diff-b20f2c72e5b61d7114744fafc5ce9934b35cc0ed7288803db8b7284639e070f6
